napari/_vispy/vispy_surface_layer.py

```diff
@@ -0,0 +1,68 @@
+from vispy.scene.visuals import Mesh as MeshNode
+from .vispy_base_layer import VispyBaseLayer
+import numpy as np
+
+
+class VispySurfaceLayer(VispyBaseLayer):
+    """Vispy view for the surface layer.
+
+    View is based on the vispy mesh node and uses default values for
+    lighting direction and lighting color. More information can be found
+    here https://github.com/vispy/vispy/blob/master/vispy/visuals/mesh.py
+    """
+
+    def __init__(self, layer):
+        node = MeshNode()
+
+        super().__init__(layer, node)
+
+        self.layer.events.colormap.connect(
+            lambda e: self._on_colormap_change()
+        )
+        self.layer.events.contrast_limits.connect(
+            lambda e: self._on_contrast_limits_change()
+        )
+        self.layer.dims.events.ndisplay.connect(
+            lambda e: self._on_display_change()
+        )
+        self.reset()
+        self._on_display_change()
+
+    def _on_display_change(self):
+        self.layer._update_dims()
+        self.layer._set_view_slice()
+        self.reset()
+        for b in range(self.layer.dims.ndisplay):
+            self.node.bounds(b)
+
+    def _on_data_change(self):
+        if len(self.layer._data_view) == 0 or len(self.layer._view_faces) == 0:
+            vertices = None
+            faces = None
+            vertex_values = np.array([0])
+        else:
+            vertices = self.layer._data_view[:, ::-1] + 0.5
```

Comment:
Can you add a note here about the offset? eg "vispy places pixel coordinates at the corners, with pixel centers offset at 0.5, but for our data we place values at the pixel centers. Here we convert from data space to vispy space"

Maybe there's a way to make that more concise. =P

Reply:
added a shortened version, we might want to put this message on the other layers too, but i'll just put it here for now